File: DEVBGJAVADEMO/DEVBGJAVADEMO/src/test/java/steps/LoginSteps.java
package steps;

import hooks.UiHooks;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import pages.LoginPage;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class LoginSteps {
    @When("the user logs in via the UI")
    public void theUserLogsInViaTheUI() {
        var page = new LoginPage(UiHooks.driver);
        page.open(UiHooks.BASE_UI_URL);
        page.login(UiHooks.createdEmail, UiHooks.createdPassword);
    }

    @When("logs in with {string} and {string} password via the UI")
    public void logsInWithAndPasswordViaTheUI(String username, String password) {
        var page = new LoginPage(UiHooks.driver);
        page.open(UiHooks.BASE_UI_URL);
        page.login(username, password);
    }

    @Then("the dashboard is visible")
    public void theDashboardIsVisible() {
        var page = new LoginPage(UiHooks.driver);
        assertTrue(page.isDashboardVisible());
    }
}

File: DEVBGDEMO/DEVBGDEMO/Steps/LoginSteps.cs
using Reqnroll;
using FluentAssertions;
using DEVBGDEMO.Hooks;
using DEVBGDEMO.Pages;

namespace DEVBGDEMO.Steps
{
    [Binding]
    public class LoginSteps
    {
        [When("the user logs in via the UI")]
        public void WhenUserLogsInViaUi()
        {
            var page = new LoginPage(UiHooks.Driver!);
            page.Open(UiHooks.BASE_UI_URL);
            page.Login(UiHooks.CreatedEmail!, UiHooks.CreatedPassword!);
        }

        [When(@"logs in with ""([^""]+)"" and ""([^""]+)"" password via the UI")]
        public void WhenUserLogsInViaUi(string username, string password)
        {
            var page = new LoginPage(UiHooks.Driver!);
            page.Open(UiHooks.BASE_UI_URL);
            page.Login(username, password);
        }

        [Then("the dashboard is visible")]
        public void ThenDashboardVisible()
        {
            var page = new LoginPage(UiHooks.Driver!);
            page.IsDashboardVisible().Should().BeTrue("dashboard should be visible after successful login");
        }
    }
}
